CondCore/PopCon/src/OnlinePopCon.cc

```diff
@@ -0,0 +1,116 @@
+    m_tagInfo.name = tagName;
+    if (m_targetConnectionString.empty()) {
+      m_targetSession = m_dbService->session();
+      m_dbService->startTransaction();
+    } else {
+      cond::persistency::ConnectionPool connPool;
+      connPool.setAuthenticationPath(m_authPath);
+      connPool.setAuthenticationSystem(m_authSys);
+      connPool.configure();
+      m_targetSession = connPool.createSession(m_targetConnectionString);
+      m_targetSession.transaction().start();
+    }
+    if (m_targetSession.existsDatabase() && m_targetSession.existsIov(tagName)) {
+      cond::persistency::IOVProxy iov = m_targetSession.readIov(tagName);
+      m_tagInfo.size = iov.sequenceSize();
+      if (m_tagInfo.size > 0) {
+        m_tagInfo.lastInterval = iov.getLast();
+      }
+      m_dbService->logger().logInfo() << "OnlinePopCon::preparePopCon";
+      m_dbService->logger().logInfo() << "  destination DB: " << connectionStr;
+      m_dbService->logger().logInfo() << "  target DB: "
+                                      << (m_targetConnectionString.empty() ? connectionStr : m_targetConnectionString);
+      m_dbService->logger().logInfo() << "  TAG: " << tagName << ", last since/till: " << m_tagInfo.lastInterval.since
+                                      << "/" << m_tagInfo.lastInterval.till;
+      m_dbService->logger().logInfo() << "  size: " << m_tagInfo.size;
+    } else {
+      m_dbService->logger().logInfo() << "OnlinePopCon::preparePopCon";
+      m_dbService->logger().logInfo() << "  destination DB: " << connectionStr;
+      m_dbService->logger().logInfo() << "  target DB: "
+                                      << (m_targetConnectionString.empty() ? connectionStr : m_targetConnectionString);
+      m_dbService->logger().logInfo() << "  TAG: " << tagName << "; First writer to this new tag.";
+    }
+    return m_targetSession;
+  }
+
+  cond::persistency::Session OnlinePopCon::initialize() {
+    // Check if DB service is available
+    if (!m_dbService.isAvailable()) {
+      throw Exception("OnlinePopCon", "[initialize] DBService not available");
+    }
+
+    // Start DB logging service
+    m_dbLoggerReturn_ = 0;
+    m_dbService->logger().start();
+    m_dbService->logger().logInfo() << "OnlinePopCon::initialize - begin logging for record: " << m_recordName;
+
+    // If requested, lock records
+    if (m_useLockRecors) {
+      m_dbService->logger().logInfo() << "OnlinePopCon::finalize - locking records";
+      m_dbService->lockRecords();
+    }
+
+    // Prepare the rest of PopCon infrastructure
+    auto session = preparePopCon();
+    return session;
+  }
+
+  void OnlinePopCon::finalize() {
+    // Check if DB service is available
+    if (!m_dbService.isAvailable()) {
+      throw Exception("OnlinePopCon", "[finalize] DBService not available");
+    }
+
+    // Release locks if previously locked
+    if (m_useLockRecors) {
+      m_dbService->logger().logInfo() << "OnlinePopCon::finalize - releasing locks";
```

Comment:
probably you meant "initialize" instead of "finalize"

Reply:
ah ok you commented on the wrong line ehehe you are right, I will fix line 83